### PR TITLE
Login refactor

### PR DESCRIPTION
**Pull request by nickcobbett**

### CODE REVIEW

File: index.ios.js [hunk 1]
```diff
@@ -23,23 +23,68 @@ import Replay from './src/components/Replay';
...
+          profile: profile,
+          userId: profile.identities[0].userId
+        });
+        console.log('this.state.profile', this.state.profile);
+        console.log('this.state.userId', this.state.userId)
+
+      }
+    })
+
+    // authorizeUser((err, profile) => {
```
Comment (jindac) [inline index.ios.js]: Is all this code in index.ios.js necessary?

File: index.ios.js [hunk 2]
```diff
@@ -23,23 +23,32 @@ import Replay from './src/components/Replay';
 
 import {findDistance, processLocation, getRaceStatus} from './src/utils/raceUtils';
 import race from './assets/presetChallenges/standardWalk.json';
-import {checkStorage, logOutUser} from './src/utils/loginUtils.js';
+import {checkStorage, logOutUser, loginUser, authorizeUser} from './src/utils/loginUtils.js';
```
Comment (JamesFalkoff) [inline index.ios.js:26]: It appears we only need to import loginUser based on the current code in componentWillMount

File: src/utils/loginUtils.js [hunk 3]
```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
...
+//     callback(profile)
+//   })
+// }
 
-export function saveProfileAndToken(profile, token) {
-  AsyncStorage.setItem('profile', JSON.stringify(profile), (err, success) => {
+// export function saveProfileAndToken(profile, token, callback) {
+//   AsyncStorage.setItem('profile', JSON.stringify(profile), (err, success) => {
+//     if (err) {
+//       console.log('loginUtils -> saveProfileAndToken -> profile', err);
```
Comment (JamesFalkoff) [inline src/utils/loginUtils.js:31]: When we get an error (here and elsewhere), we should probably call the callback with callback(err, null)

File: src/utils/loginUtils.js [hunk 4]
```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
...
+//       'Accept': 'application/json',
+//       'Content-Type': 'application/json'
+//     },
+//     body: JSON.stringify(nameAndId)
+//   }).then(response => {
+//     console.log('loginUtils -> verifyProfile ', response);
+//     callback(null, profile)
+//   }).catch(err => {
+//     console.log('loginUtils -> verifyProfile error', err)
+//     callback(profile)
```
Comment (JamesFalkoff) [inline src/utils/loginUtils.js:24]: Not the right number of arguments for the callback?

File: src/utils/loginUtils.js [hunk 5]
```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
...
-      console.log('Profile Removed');
-      loginUser();
+      saveProfileAndToken(profile, token, callback);
     }
   });
-}
\ No newline at end of file
+}
+
+export function saveProfileAndToken(profile, token, callback) {
```
Comment (JamesFalkoff) [inline src/utils/loginUtils.js:140]: It looks like this is overwriting the other implementation of saveProfileAndToken in this same file?

File: src/utils/loginUtils.js [hunk 6]
```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
...
+  AsyncStorage.setItem('profile', JSON.stringify(profile), (err, success) => {
+    if (err) {
+      console.log('loginUtils -> saveProfileAndToken -> profile', err);
```
Comment (JamesFalkoff) [inline src/utils/loginUtils.js:143]: As is, the callback will not be called if we get an error. Should probably call it here with callback(err, null).

File: src/utils/loginUtils.js [hunk 7]
```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
...
+//       })
+//     }
+//   })
+// }
+
+// export function loginUser(callback) {
+//   var lock = new Auth0Lock(facebookKey);
+//   lock.show({}, (err, profile, token) => {
+//     if (err) {
+//       console.log('loginUtils -> loginUser -> error', err);
```
Comment (JamesFalkoff) [inline src/utils/loginUtils.js:52]: If error, callback never gets called

**Pull request merged**